torchaudio/prototype/io/streamer.py

```diff
@@ -479,19 +488,30 @@ def fill_buffer(self) -> int:
                 this method.
         """
         while not self.is_buffer_ready():
-            for _ in range(3):
-                code = self.process_packet()
-                if code != 0:
-                    return code
+            code = self.process_packet(retry)
+            if code != 0:
+                return code
         return 0
 
-    def stream(self) -> Iterator[Tuple[Optional[torch.Tensor]]]:
-        """Return an iterator that generates output tensors"""
+    def stream(self, retry: Optional[float] = 10) -> Iterator[Tuple[Optional[torch.Tensor]]]:
+        """Return an iterator that generates output tensors
+
+        Arguments:
+            retry (float or None, optional): See :py:func:`~Streamer.process_packet`.
+
+        Returns:
+            Iterator[Tuple[Optional[torch.Tensor]]]:
```

Comment:
is it the case that each tuple returned by the function contains just one element that's either a tensor or `None`?

Reply:
Yeah, it starts from the list of tuple of tensors, each corresponds to one output stream, then as output streams get exhausted, they start to return None. It will stop when all the output streams get exhausted.

Reply:
ah i see. then, we should amend the type hint
```suggestion
            Iterator[Tuple[Optional[torch.Tensor], ...]]:
```